.travis.yml

```diff
@@ -1,48 +1,58 @@
 language: php
-php:
-  - 7.1
-  - 7.2
-  - nightly
+php: [ 7.1, 7.2, nightly ]
 sudo: false
 
 env:
+
 matrix:
   fast_finish: true
   allow_failures:
     - php: nightly
 
 install:
   - composer install --no-interaction --prefer-dist --optimize-autoloader
+install:
+  - travis_retry composer install --no-interaction --prefer-dist --optimize-autoloader
+  - travis_retry composer global require phpunit/phpunit ^6 # cannot use phpunit.phar or require-dev, because this package is a phpunit dep
+  - travis_retry wget --no-verbose https://phar.io/releases/phive.phar
+
+script:
+  - /home/travis/.composer/vendor/bin/phpunit --no-coverage
+  - bin/behat -p default
 
 jobs:
   include:
-    - stage: test
+    - stage: dogfood
+      php: 7.1
       script:
-      - bin/phpunit --no-coverage
-      - bin/behat -p default
+        - chmod u+x ./bin/phpdoc
+        - ./bin/phpdoc run
 
-    - stage: test
+    - stage: phar
```

Comment:
moving this to a separate stage doesn't really make any sense I think. build will take longer with this since it cannot start a new stage before all prior stages and jobs are done. Failing tests on a certain phpversion will hide broking things in the phar.

Reply:
I hadn't realized that separate configs could still be labeled as the same stage.